internal/provider/physical_replication_stream.go

```diff
@@ -0,0 +1,384 @@
+		StandbyClusterId: plan.StandbyClusterId.ValueString(),
+	}
+
+	traceAPICall("CreatePhysicalReplicationStream")
+	streamObj, _, err := r.provider.service.CreatePhysicalReplicationStream(ctx, &apiRequest)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating PCR stream",
+			fmt.Sprintf("Could not create PCR stream: %s", formatAPIErrorMessage(err)),
+		)
+		return
+	}
+
+	err = retry.RetryContext(
+		ctx,
+		clusterCreateTimeout,
+		waitForPhysicalReplicationStreamStatusFunc(
+			ctx, streamObj.Id, r.provider.service, streamObj, client.REPLICATIONSTREAMSTATUSTYPE_REPLICATING,
+		),
+	)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"PCR stream creation failed",
+			fmt.Sprintf("PCR stream failed to create: %s", formatAPIErrorMessage(err)),
+		)
+		return
+	}
+
+	loadPCRStreamToTerraformState(streamObj, &plan)
+	diags = resp.State.Set(ctx, plan)
+	resp.Diagnostics.Append(diags...)
+}
+
+func waitForPhysicalReplicationStreamStatusFunc(
+	ctx context.Context, id string, cl client.Service, stream *client.PhysicalReplicationStream,
+	desiredStatus client.ReplicationStreamStatusType,
+) retry.RetryFunc {
+	return func() *retry.RetryError {
+		traceAPICall("GetPhysicalReplicationStream")
+		apiStream, httpResp, err := cl.GetPhysicalReplicationStream(ctx, id)
+		if err != nil {
+			if httpResp != nil && httpResp.StatusCode < http.StatusInternalServerError {
+				return retry.NonRetryableError(fmt.Errorf("error getting physical replication stream: %s", formatAPIErrorMessage(err)))
+			} else {
+				return retry.RetryableError(fmt.Errorf("encountered a server error while reading physical replication stream status"))
+			}
+		}
+		*stream = *apiStream
+		if stream.Status != desiredStatus {
+			return retry.RetryableError(fmt.Errorf("physical replication stream has status %s; desired status is %s", stream.Status, desiredStatus))
+		}
+		return nil
+	}
+}
+
+func loadPCRStreamToTerraformState(
+	apiObj *client.PhysicalReplicationStream, state *PhysicalReplicationStream,
+) {
+	state.ID = types.StringValue(apiObj.GetId())
+	state.PrimaryClusterId = types.StringValue(apiObj.GetPrimaryClusterId())
+	state.StandbyClusterId = types.StringValue(apiObj.GetStandbyClusterId())
+	state.CreatedAt = types.StringValue(apiObj.GetCreatedAt().String())
+	if apiObj.ReplicatedTime != nil {
+		state.ReplicatedTime = types.StringValue(apiObj.GetReplicatedTime().String())
+	}
+	if apiObj.ReplicationLagSeconds != nil {
+		state.ReplicationLagSeconds = types.Int64Value(int64(apiObj.GetReplicationLagSeconds()))
+	}
+	if apiObj.RetainedTime != nil {
+		state.RetainedTime = types.StringValue(apiObj.GetRetainedTime().String())
+	}
+	if apiObj.ActivatedAt != nil {
+		state.ActivatedAt = types.StringValue(apiObj.GetActivatedAt().String())
+	} else {
+		state.ActivatedAt = types.StringNull()
+	}
+	state.Status = types.StringValue(string(apiObj.GetStatus()))
+
+	if state.FailoverImmediately.ValueBool() {
+		state.FailoverImmediately = types.BoolValue(true)
+	}
```

Comment:
This seems a bit odd.  If `state.FailoverImmediately.ValueBool()` is true, then `state.FailoverImmediately` should already equal `types.BoolValue(true)` right?  I would think this is a no-op